extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -357,9 +357,16 @@ func (a *genericActuator) waitUntilWantedMachineDeploymentsAvailable(ctx context
 				}
 			}
 
-			// If the shoot get hibernated we want to wait until all wanted machine deployments have been deleted
-			// entirely.
+			// We want to wait until all wanted machine deployments have as many
+			// available replicas as desired (specified in the .spec.replicas).
+			// However, if we see any error in the status of the deployment then we return it.
+			for _, failedMachine := range deployment.Status.FailedMachines {
```

Comment:
While you are already at it: if there are multiple failed machines, can we return a summary/or the number of all failed machines here. Otherwise only one failed machine is reported.

Reply:
Here my suggestion 9880b38ff1116397f4e6786afc9094d175071246.